### PR TITLE
ci: build inside a builder container

### PR DESCRIPTION
**Pull request by sverrehu**
Something like this, @erikgb? We will, of course, need to ditch anything maven from the actions if we select to go this way.

### CODE REVIEW

**Comment by erikgb** [comment]
> Something like this, @erikgb? We will, of course, need to ditch anything maven from the actions if we select to go this way.

Yes! ❤️ 

image

Can you remove the maven build from outside the Docker-build?

**Comment by sverrehu** [comment]
> Can you remove the maven build from outside the Docker-build?

Done. Although I'm still on thin ice when it comes to GitHub actions, so not sure it is correct.

**Comment by erikgb** [comment]
We should also probably test changing `pom.xml`. That should invalidate the Docker cache from the copy `pom.xml` statement. I expect that to be the same as the GH Maven cache. We should not have SNAPSHOT-dependencies. If we did, those would be problematic - since they will not invalidate the cache.

**Comment by sverrehu** [comment]
> We should also probably test changing `pom.xml`.

I tested locally, first changing the pom and then a source file. The results were as we hope for.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,12 @@
+FROM maven:3-eclipse-temurin-17 AS builder
```
Comment (erikgb) [inline Dockerfile:1]:
```suggestion
FROM maven:3-eclipse-temurin-17-alpine AS builder
```

I expect the image to be slightly smaller, but Java is the big fat thing here. anyway...

**Comment by erikgb** [comment]
> > We should also probably test changing `pom.xml`.
> 
> I tested locally, first changing the pom and then a source file. The results were as we hope for. But I'm only using `docker build`, and nothing fancy.

The GH Docker cache setup is quite aggressive, but this seems to work as we want.

**Comment by sverrehu** [comment]
> The GH Docker cache setup is quite aggressive, but this seems to work as we want.

There are commits for "change pom" and "change source file" that probably have some action runs.

**Review by erikgb — APPROVED** [review]
LGTM! Let's try this for some time!

**Pull request merged**